src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchWriter.cs

```diff
@@ -83,6 +83,11 @@ internal sealed class ODataJsonLightBatchWriter : ODataBatchWriter
         /// </summary>
         private readonly IJsonWriter jsonWriter;
 
+        /// <summary>
+        /// The underlying asynchronous JSON writer.
+        /// </summary>
+        private readonly IJsonWriterAsync asynchronousJsonWriter;

```

Comment:
By the time we were having the discussion, I had already merged the PR that introduced this interface. I therefore decided to waiting for a major release opportunity to do the clean up. In any case, when the major release opportunity presents itself, `IJsonWriterAsync` will be merged into `IJsonWriter`. We only introduced `IJsonWriterAsync` to prevent a breaking change since `IJsonWriter` is a public interface.